src/test/libslic3r/test_support_material.cpp

```diff
@@ -0,0 +1,295 @@
+#include <catch.hpp>
+#include <libslic3r/IO.hpp>
+#include <libslic3r/GCodeReader.hpp>
+
+#include "libslic3r.h"
+#include "TriangleMesh.hpp"
+#include "Model.hpp"
+#include "SupportMaterial.hpp"
+
+using namespace std;
+using namespace Slic3r;
+
+void test_1_checks(Print &print, bool &a, bool &b, bool &c, bool &d);
+bool test_6_checks(Print &print);
+
+// Testing 0.1: supports material member functions.
+TEST_CASE("", "")
+{
+    // Create a mesh & modelObject.
+    TriangleMesh mesh = TriangleMesh::make_cube(20, 20, 20);
+
+    // Create modelObject.
+    Model model = Model();
+    ModelObject *object = model.add_object();
+    object->add_volume(mesh);
+    model.add_default_instances();
+
+    // Align to origin.
+    model.align_instances_to_origin();
+
+    // Create Print.
+    Print print = Print();
+    vector<coordf_t> contact_z = {1.9};
+    vector<coordf_t> top_z = {1.1};
+    print.default_object_config.support_material = 1;
+    print.default_object_config.set_deserialize("raft_layers", "3");
+    print.add_model_object(model.objects[0]);
+    print.objects.front()->_slice();
```

Comment:
I suggest calling ->slice() instead of _slice().

Reply:
Ok, LoH I will replace it :),